lib/handlebars/runtime.js

```diff
@@ -197,7 +197,12 @@ export function wrapProgram(container, i, fn, data, declaredBlockParams, blockPa
 export function resolvePartial(partial, context, options) {
   if (!partial) {
     if (options.name === '@partial-block') {
-      partial = options.data['partial-block'];
+      let data = options.data;
+      while (data['partial-block'] === noop) {
```

Comment:
After I set `partial` to `data['partial-block']`, I set `data['partial-block']` to `noop` on line 205 so that we'll know it's been used the next time we're in here.